### PR TITLE
IE11 compatibility fixes

### LINKED ISSUE #4470: Dialogs are shifted in IE11
Metabase 0.23-snapshot
IE 11

Dialogs are shifted:
image

### LINKED ISSUE #4471: No save button when user try to add a card on dashboard in IE11
Metabase 0.23-snapshot
IE11

No save button when I try to add a card on dashboard:
image

### PR DESCRIPTION
**Pull request by attekei**
- [x] Remove `margin: auto` from modals when using flexbox, fixes #4470
- [x] Add `flex-basis-auto` class for fixing situations where flex item doesn't grow with its content although growing is a desired behavior. The ModalBody component needed this. 
- [x] Fix error message overflowing issue in LoadingAndErrorWrapper
- [x] Fix broken Save button position in dashboard edit bar
- [x] Fix revision history modal height
- [x] Show tutorial arrow above other elements (there are some other tutorial UI bugs as well related to `<Portal>` element but couldn't fix those easily; it's possible to complete tutorial anyway)
- [x] Fix the display of "Preview SQL" modal (this was broken on all browsers)
- [x] Don't show the clear button on IE11 inputs because it doesn't trigger the React onInput event
- [x] Fix blank full-screen dashboard by giving to body element `width: 100%` 
- [x] Fix the centering bug of sharing/embedding page

- [x] Fix #4471

All new CSS classes are extensively commented.

### CODE REVIEW

File: frontend/src/metabase/css/core/flex.css [hunk 1]
```diff
@@ -16,7 +16,22 @@
...
+ * browsers. Specifically:
+ * - On Chrome and FF it's set to `flex-basis: 0%`. That behaves equally as `height: 0%`.
+ *   It means that if the containing block has no explicit height, then `height: 0%` is computed as `height: auto`,
+ *   and element grows as its content grows. That is the most common scenario in Metabase codebase.
+ * - On older IEs it's set to `flex-basis: 0` which means that the initial main size of flex item zero.
+ *   It's also notable that `flex-basis: 0%` doesn't work correctly on IE.
+ *
+ *  As a solution, `flex-basis-auto` should always be used in conjunction with `flex-full` when it is
+ *  a desired behavior that the element grows with its contents.
+*/
```
Comment (salsakran) [inline frontend/src/metabase/css/core/flex.css]: 👍

**Comment by attekei** [comment]
I think that this could already be reviewed and merged, and the fix for #4471 could go to a separate PR.

**Review by kdoh — APPROVED** [review]
Dig the comments. 👍

**Comment by attekei** [comment]
Fixed #4471 as well, now things should look pretty good in the IE11 world.

File: frontend/src/metabase/query_builder/components/QueryModeButton.jsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ export default class QueryModeButton extends Component {
                     </span>
                 </Tooltip>
 
-                <Modal medium backdropClassName="Modal-backdrop--dark" isOpen={this.state.isOpen} onClose={() => this.setState({isOpen: false})}>
+                <Modal medium backdropClassName="Modal-backdrop-dark" isOpen={this.state.isOpen} onClose={() => this.setState({isOpen: false})}>
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/QueryModeButton.jsx]: Following "BEM" this was actually correct and it's the `.Modal-backdrop-dark` selector that should be changed to `.Modal-backdrop--dark`

Comment (attekei) [inline frontend/src/metabase/query_builder/components/QueryModeButton.jsx]: will switch to `.Modal-backdrop--dark`

**Pull request merged**